components/context_id/src/lib.rs

```diff
@@ -109,34 +109,39 @@ impl ContextIDComponentInner {
             },
         };
 
-        let (creation_timestamp, generated_creation_timestamp) = if generated_context_id {
-            // If we had to generate a UUID, also force a new timestamp.
-            (now, true)
-        } else {
-            match creation_timestamp_s {
-                secs if secs > 0 => (
-                    DateTime::<Utc>::from_timestamp(secs, 0)
-                        .ok_or(Error::InvalidTimestamp { timestamp: secs })?,
+        let (creation_timestamp, generated_creation_timestamp, force_rotation) =
+            match (generated_context_id, creation_timestamp_s) {
+                // We generated a new context ID then we need a fresh timestamp and no rotation needed
+                (true, _) => (now, true, false),
+                // Pre-existing context ID with valid timestamp then use timestamp and no rotation needed
+                (false, secs) if secs > 0 => (
+                    DateTime::<Utc>::from_timestamp(secs, 0).unwrap_or(now),
```

Comment:
If we fail to unwrap this, then we use `now`, and then we _should_ force rotation though, right? We have no idea how old the context ID is, so we should probably just generate a new one and invalidate the old one... or is there no way that `from_timestamp(secs, 0)` can fail to unwrap with a positive i64?

Reply:
So according to [the docs](https://docs.rs/chrono/latest/chrono/naive/struct.NaiveDateTime.html#panics) it will only use fail if we exceed a limit of 260, 000 years from CE; effectively never for any real UNIX epoch. With that being said, this is the stuff that one thinks would never happen but it does 😂 I will update to force a rotation when we use now in that branch too. Updated in [7ab66bb](https://github.com/mozilla/application-services/pull/6787/commits/7ab66bb6cd260763aaeca65d55c13fffbb3ead3a)